test/resource.cpp

```diff
@@ -35,8 +35,8 @@ TEST(CampResource, ConvertFails)
 {
   Resource h1{Host()};
   h1.get<Host>();
-  ASSERT_THROW(h1.get<Host2>(), std::runtime_error);
-  ASSERT_FALSE(h1.try_get<Host2>());
+  //ASSERT_THROW(h1.get<Host2>(), std::runtime_error);
```

Comment:
If you are going to leave this in and commented out, please add a comment why it is commented out.

Reply:
This was put back in and the problematic operator== functions removed